### PR TITLE
Database display added; includes important updates to databaseController

### PR DESCRIPTION
**Pull request by michibatoryu**

### CODE REVIEW

File: src/main/java/Database/DatabaseController.java [hunk 1]
```diff
@@ -1,90 +1,35 @@
...
 import java.sql.*;
 import java.util.ArrayList;
+import java.util.LinkedList;
 import java.util.List;
-import java.util.Scanner;
+import org.json.JSONObject;
 
-public class DatabaseController {
-  public Connection getC() {
-    return c;
```
Comment (KalticCodes) [inline src/main/java/Database/DatabaseController.java]: manual made constructor vs lombok; irrelevant stylistic choice

File: src/main/java/edu/wpi/teame/controllers/DatabaseViewController.java [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+
+    longNameCol.setCellValueFactory(new PropertyValueFactory<LocationName, String>("longName"));
+    shortNameCol.setCellValueFactory(new PropertyValueFactory<LocationName, String>("shortName"));
+    nodeTypeCol.setCellValueFactory(new PropertyValueFactory<LocationName, String>("nodeType"));
+
+    ObservableList locationList = FXCollections.observableArrayList(dC.getLocationName());
+    // ArrayList<LocationName> locationArrList = new ArrayList<>();
+    // locationArrList.add(new LocationName("test long", "testshort", LocationName.NodeType.INFO));
+    // locationArrList.add(new LocationName("test looong", "tstshrt", LocationName.NodeType.EXIT));
+    // ObservableList locationList = FXCollections.observableArrayList(locationArrList);
```
Comment (KalticCodes) [inline src/main/java/edu/wpi/teame/controllers/DatabaseViewController.java]: why is this code commented? if it was just for testing please remove it (applies to all commented code)

File: src/main/java/edu/wpi/teame/controllers/DatabaseViewController.java [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+                  "Move", selectedFile.getParentFile().getAbsolutePath(), selectedFile.getName());
+            } catch (SQLException | IOException e) {
+              System.out.println("You messed up big time!!!!!!");
+              // System.out.println(e);
+            }
+          }
+        });
+  }
+
+  private void switchActiveTable(String db) {
```
Comment (KalticCodes) [inline src/main/java/edu/wpi/teame/controllers/DatabaseViewController.java]: why not use the enum here?

File: src/main/java/edu/wpi/teame/controllers/DatabaseViewController.java [hunk 4]
```diff
@@ -0,0 +1,433 @@
...
+        nodeAddZone.setVisible(false);
+        edgeAddZone.setVisible(true);
+        break;
+    }
+    if (!addButton.isVisible()) {
+      addButton.setVisible(true);
+    }
+  }
+
+  private void addRow(DatabaseController dC, Popup windowPop) {
```
Comment (KalticCodes) [inline src/main/java/edu/wpi/teame/controllers/DatabaseViewController.java]: why does this need to take in a DatabaseController here?

File: src/main/java/edu/wpi/teame/controllers/DatabaseViewController.java [hunk 5]
```diff
@@ -0,0 +1,433 @@
...
+    Object toAdd;
+    switch (databaseChoice.getValue()) {
```
Comment (KalticCodes) [inline src/main/java/edu/wpi/teame/controllers/DatabaseViewController.java:374]: why does databaseChoice not use the enum?

**Review by KalticCodes — CHANGES_REQUESTED** [review]
Importing from a CSV file causes the following error
image

Getting "No Content In Table" message for all tables but "move".

Delete row functionality does not seem to work if more than just the Node ID text box is filled

Add functionality also doesn't seem to work consistently; Getting an "Improper formatting" error.

Entering the Database screen takes a long time, and sometimes causes the program to not respond or crash.

Database Menu button does not highlight on hover, which is different than the rest of the buttons

File: src/main/java/edu/wpi/teame/controllers/DatabaseViewController.java (hunk 4 above)
Comment (michibatoryu) [reply src/main/java/edu/wpi/teame/controllers/DatabaseViewController.java]: a scope thing; dC isn't declared globally for the class (same for the Popup)

File: src/main/java/edu/wpi/teame/controllers/DatabaseViewController.java (hunk 2 above)
Comment (michibatoryu) [reply src/main/java/edu/wpi/teame/controllers/DatabaseViewController.java]: yeah those were for testing - removed them

Comment (KalticCodes) [reply src/main/java/edu/wpi/teame/controllers/DatabaseViewController.java]: Sounds good; resolving

File: src/main/java/edu/wpi/teame/controllers/DatabaseViewController.java (hunk 5 above)
Comment (michibatoryu) [reply src/main/java/edu/wpi/teame/controllers/DatabaseViewController.java:374]: true, will change that

File: src/main/java/edu/wpi/teame/controllers/DatabaseViewController.java (hunk 4 above)
Comment (KalticCodes) [reply src/main/java/edu/wpi/teame/controllers/DatabaseViewController.java]: why note just use DatabaseController.INSTANCE?

Comment (michibatoryu) [reply src/main/java/edu/wpi/teame/controllers/DatabaseViewController.java]: ooooooh I forgot that that was a thing that worked now I'll do that

File: src/main/java/Database/DatabaseController.java [hunk 6]
```diff
@@ -9,6 +9,10 @@
 import java.util.List;
 import org.json.JSONObject;
 
+import static edu.wpi.teame.map.Floor.floorToString;
+import static edu.wpi.teame.map.Floor.stringToFloor;
+import static edu.wpi.teame.map.LocationName.NodeType.stringToNodeType;
```
Comment (KalticCodes) [inline src/main/java/Database/DatabaseController.java]: if you import teame.map.Floor, you should automatically get the enum and the methods associate with it. Same thing for teame.map.LocationName.NodeType

File: src/main/java/Database/DatabaseController.java [hunk 7]
```diff
@@ -239,22 +256,20 @@ public List<MoveAttribute> getMoveList() {
       System.out.println("Move table retrieved successfully");
     }
     return moveList;
-  }
+  }*/
```
Comment (KalticCodes) [inline src/main/java/Database/DatabaseController.java]: Commented out code

File: src/main/java/Database/DatabaseController.java [hunk 8]
```diff
@@ -239,22 +256,20 @@ public List<MoveAttribute> getMoveList() {
...
 
   public List<HospitalEdge> getEdges() {
```
Comment (KalticCodes) [inline src/main/java/Database/DatabaseController.java:238]: not a now thing, but could be worth looking into a List<Object> getFromTable(Table table) method for the future

File: src/main/java/edu/wpi/teame/controllers/DatabaseViewController.java [hunk 9]
```diff
@@ -98,19 +98,18 @@ public class DatabaseViewController {
...
-    ArrayList<String> choices = new ArrayList<String>();
-    choices.add("move");
-    choices.add("location");
-    choices.add("node");
-    choices.add("edge");
+    ArrayList<DatabaseController.Table> choices = new ArrayList<>();
+    choices.add(DatabaseController.Table.MOVE);
+    choices.add(DatabaseController.Table.LOCATION_NAME);
+    choices.add(DatabaseController.Table.NODE);
+    choices.add(DatabaseController.Table.EDGE);
```
Comment (KalticCodes) [inline src/main/java/edu/wpi/teame/controllers/DatabaseViewController.java:115]: For the future, I think there may be an easy way to do Enum->ObservableArray? not completely sure

File: src/main/java/edu/wpi/teame/controllers/DatabaseViewController.java (hunk 5 above)
Comment (KalticCodes) [reply src/main/java/edu/wpi/teame/controllers/DatabaseViewController.java:374]: Confirmed, resolving

File: src/main/java/Database/DatabaseController.java [hunk 10]
```diff
@@ -309,10 +294,12 @@ public List<LocationName> getLocationName() {
       String sql = "SELECT \"longName\", \"shortName\", \"nodeType\" FROM teame.\"LocationName\";";
       ResultSet rs = stmt.executeQuery(sql);
 
-
       while (rs.next()) {
-        locationNames.add(new LocationName(rs.getString("longName") + "", rs.getString("shortName"), stringToNodeType(rs.getString("nodeType"))));
-
+        locationNames.add(
+            new LocationName(
+                rs.getString("longName") + "",
```
Comment (KalticCodes) [inline src/main/java/Database/DatabaseController.java]: + "" not needed here, getString should already return a string

**Review by KalticCodes — CHANGES_REQUESTED** [review]
Couple different issues associated with importing from CSV

image
image

Database display page is off center compared to the rest of the application; opening the page from the default window size results in the page being basically blank, with the database selection combobox being half cut off. Maybe add some indication that the user needs to increase the size of the screen?

File: src/main/java/Database/DatabaseController.java [hunk 11]
```diff
@@ -41,17 +42,21 @@ public static String tableToString(Table tb) {
     }
   }
 
+  public static void main(String[] args) {
```
Comment (KalticCodes) [inline src/main/java/Database/DatabaseController.java]: do we need this main method here? if it's just for testing make it a test case or remove it

File: src/main/java/Database/DatabaseController.java [hunk 12]
```diff
@@ -64,7 +69,7 @@ private Connection connectToDatabase(String username, String password) {
       System.err.println(e.getClass().getName() + ": " + e.getMessage());
       System.exit(0);
     }
-    System.out.println("Opened database successfully");
+    // System.out.println("Opened database successfully");
```
Comment (KalticCodes) [inline src/main/java/Database/DatabaseController.java]: Why comment this status report?

**Review by KalticCodes — CHANGES_REQUESTED** [review]
Just a few small issues, all behavior confirmed

**Review by KalticCodes — APPROVED** [review]
All behavior confirmed, all code reviewed. Approving

**Pull request merged**